lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,151 @@
+from http import HTTPStatus
+from logging import Logger
+
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    JSON,
+    Schema,
+    Parameter,
+    ParamType,
+)
+from pbench.server.api.resources.query_apis import CONTEXT
+from pbench.server.api.resources.query_apis.datasets import RunIdBase
+
+
+class TableOfContent(RunIdBase):
+    """
+    Table of Content API returns the list of sub-directories and files
+    present under a directory.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("run_id", ParamType.STRING, required=True),
+                Parameter("parent", ParamType.STRING, required=True),
+            ),
+        )
+
+    def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
+        """
+        Construct a pbench Elasticsearch query for getting a list of
+        documents which contains the user provided parent with files
+        and its sub-directories with metadata of run-toc index document
+        that belong to the given run id.
+
+        Args:
+            json_data: JSON dictionary of type-normalized parameters
+                "run_id": Dataset document ID
+
+        EXAMPLE:
+        {
+            "run_id": "1234567890"
+            "parent": '/default'
+        }
+        """
+        parent = json_data.get("parent")
+        run_id = context["run_id"]
```

Comment:
This is somewhat messy. The `"run_id"` parameter is required, but it's consumed by the superclass and accessed implicitly in this class through `CONTEXT`. So documenting that it's required, and including it in the example, is good. And, yes, a note in the docstring to that effect wouldn't be a bad idea.

Reply:
I suggest that we have the docstring for this function document the requirements _of this function_ and not try to cover those of the RESTful API that it is used to implement, etc.

I think that having examples is very helpful (if they are properly representative and maintained...), but they don't need to be complete -- in particular, the example for the `json_data` doesn't need to include the `"run_id"` value, since this function pulls that from the `context` parameter.

(I think it would be best to be consistent about the source of the `"run_id"` parameter -- since the other `assemble()` methods pull it from the `context`, I think we should do so here.)

Reply:
Also, it is being used in the query, It is `matching` the `run_data_parent` id.

Reply:
Yes...and I think the docstring describes that....